src/hal/i_components/ddt.icomp

```diff
@@ -19,12 +19,12 @@ component ddt "Compute the derivative of the input function";
 pin in float in;
 pin out float out;
 
-variable double old;
+variable hal_float_t old;
 
 function _;
 license "GPL";
 ;;
-double tmp = in;
+hal_float_t tmp = in;
 out = (tmp - old) / (period * 1e-9);
```

Comment:
period is implicitly converted from long to floating point variable, typecast to hal_float_t should be added